CMakeLists.txt

```diff
@@ -24,6 +24,15 @@ else()
     set(VERSION_FULL "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 endif()
 
+
+# Options
+option(ENABLE_WMAX   "Enable `-Werror`, `-Wextra`, and other compilation flags." Off)
+option(ENABLE_WERROR "Enable `-Werror` compilation flag." ${ENABLE_WMAX})
+option(ENABLE_WEXTRA "Enable `-Wextra` compilation flag." ${ENABLE_WMAX})
```

Comment:
Using `${ENABLE_WMAX}` in this manner only works as expected the first time the project is configured. For example if you do
```
cmake -DCMAKE_BUILD_TYPE=Release ..
cmake -DENABLE_WMAX=On ..
```
Neither `ENABLE_WERROR` nor `ENABLE_WEXTRA` will be `On`. imo should consider just removing `ENABLE_WMAX`

Reply:
I agree. This should either be removed or made to work. I'll go with removal.

Reply:
Should have been `Off`. Corrected.